src/app/groups/containers/group-edit/group-edit.component.ts

```diff
@@ -172,4 +213,21 @@ export class GroupEditComponent implements OnInit, OnDestroy, PendingChangesComp
   refreshGroup(): void {
     this.groupDataSource.refetchGroup();
   }
+
+  updateRequireLockMembershipApprovalUntilValueAndValidity(): void {
+    const control = this.groupForm.get('requireLockMembershipApprovalUntil');
+    if (!control) throw new Error('Unexpected: Missed requireLockMembershipApprovalUntil control');
+    const initialRequireLockMembershipApprovalUntil = this.initialFormData?.requireLockMembershipApprovalUntil;
+    if (control.value !== null && initialRequireLockMembershipApprovalUntil) {
+      const currentDate = new Date();
+      this.minLockMembershipApprovalUntilDate = initialRequireLockMembershipApprovalUntil < currentDate
+        ? initialRequireLockMembershipApprovalUntil : currentDate;
+      // eslint-disable-next-line @typescript-eslint/unbound-method
```

Comment:
Could be remove after merge with https://github.com/France-ioi/AlgoreaFrontend/pull/1697/files#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R106